File: Daily Challanges/2026-03-March/2026-03-20.py
def get_shadow(time):
    hour = int(time.split(":")[0])
    is_half = time.split(":")[1] == "30"
    length = abs(hour - 12 + 1 / 2 * is_half) ** 3
    length = length if is_half else int(length)
    direction = "east" if hour > 12 else "west"

    if hour < 6 or hour >= 18 or hour == 12:
        return "No shadow"
    else:
        return f"{length}ft {direction}"


print(get_shadow("10:00"))
print(get_shadow("15:00"))
print(get_shadow("17:30"))
print(get_shadow("06:00"))
print(get_shadow("12:00"))
print(get_shadow("18:00"))
print(get_shadow("00:00"))

File: Daily Challanges/2026-03-March/2026-03-20.js
function getShadow(time) {
  const [hours, minutes] = time.split(":").map((item) => Number(item));
  if (hours < 6 || hours >= 18 || hours == 12) {
    return "No shadow";
  }
  const h = hours + minutes / 60;
  const length = Math.abs(h - 12) ** 3;
  const direction = hours < 12 ? "west" : "east";
  return `${length}ft ${direction}`;
}

console.log(getShadow("10:00"));
console.log(getShadow("17:30"));
